### PR TITLE
Add Support for creating PPPC & System Extensions and update Exports and ESlint

### PR DESCRIPTION
**Pull request by santese**
-Added Support for creating PPPC MDM Config
-Added Support for creating System Extension MDM Config
- Update to Pliancy ESLint Config
- Move Types to Separate file
- Export Types and Remove Default Export of Addigy Class

### CODE REVIEW

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -15,20 +15,25 @@ jobs:
       - uses: actions/setup-node@v1
         with:
           node-version: 12
-      - run: npm ci
-      - run: npm test
+      - uses: actions/setup-node@v2
+        with:
+          registry-url: 'https://npm.pkg.github.com'
```
Comment (CalebAlbers) [inline .github/workflows/npm-publish.yml:20]: Let's publish to NPM instead of GH Packages for this one

File: package.json [hunk 2]
```diff
@@ -33,6 +38,7 @@
     "url": "https://github.com/calebalbers/addigy-node"
   },
   "dependencies": {
+    "@pliancy/eslint-config-ts": "0.0.5",
```
Comment (CalebAlbers) [inline package.json:41]: We can move this and Typescript to dev deps

**Review by CalebAlbers — CHANGES_REQUESTED** [review]
Nice work adding the PPPCs, system extensions, and thanks for updating eslint.

Can you write a quick example in the README for creating PPPCs, Sys Extensions, and KEXTs? Given we're using an interface that is modified (normalized) from what Addigy uses, it'll be good to provide docs for people on usage.

File: package.json (hunk 2 above)
Comment (santese) [reply package.json:41]: > Nice work adding the PPPCs, system extensions, and thanks for updating eslint.
> 
> Can you write a quick example in the README for creating PPPCs, Sys Extensions, and KEXTs? Given we're using an interface that is modified (normalized) from what Addigy uses, it'll be good to provide docs for people on usage.

Yeah will do and I'll also add some decorators as well

File: package.json [hunk 3]
```diff
@@ -1,67 +1,74 @@
...
+        "@types/jest": "^26.0.21",
+        "@types/node": "^14.14.6",
+        "@types/uuid": "^7.0.3",
+        "@typescript-eslint/eslint-plugin": "^4.22.1",
+        "@typescript-eslint/parser": "^4.22.1",
+        "cpy-cli": "^3.1.1",
+        "eslint": "^7.12.1",
+        "eslint-config-prettier": "8.1.0",
+        "eslint-plugin-prettier": "^3.1.4",
+        "eslint-plugin-unused-imports": "^1.0.1",
```
Comment (CalebAlbers) [inline package.json:57]: The ESLint dev deps can be removed - they are child deps of the pliancy ESLint onfig. Same with `prettier`

**Review by CalebAlbers — COMMENTED** [review]
Thank you for writing the docs on the new commands and being through on the JSDoc side as well for those! I added a note about removing ESLint dev deps.

Great work on this, as always! Feel free to merge and publish a major release.

**Comment by santese** [comment]
Update to 2.0.0

**Pull request merged**